core/dbt/cli/params.py

```diff
@@ -274,14 +264,22 @@
     default=(),
 )
 
-select = click.option(
-    "-s",
-    "--select",
-    envvar=None,
-    help="Specify the nodes to include.",
-    cls=MultiOption,
-    type=tuple,
-)
+model_decls = ("-m", "--models", "--model")
+select_decls = ("-s", "--select")
+select_attrs = {
+    "envvar": None,
+    "help": "Specify the nodes to include.",
+    "cls": MultiOption,
+    "type": tuple,
+}
+
+# `--select` and `--models` are analogous for most commands except `dbt list` for legacy reasons.
+# Most CLI arguments should use the combined `select` option that aliases `--models` to `--select`.
+# However, if you need to split out these separators (like `dbt ls`), use the `models` and `raw_select` options instead.
+# See https://github.com/dbt-labs/dbt-core/pull/6774#issuecomment-1408476095 for more info.
+models = click.option(*model_decls, **select_attrs)
+raw_select = click.option(*select_decls, **select_attrs)
+select = click.option(*select_decls, *model_decls, **select_attrs)
```

Comment:
Nit: this is simpler code but slightly harder to read compared to just write the options here.